resolving/npm-resolver/src/index.ts

```diff
@@ -404,3 +527,17 @@ function getIntegrity (dist: {
   }
   return integrity.toString()
 }
+
+function createVersionSpec (version: string, pinnedVersion?: PinnedVersion): string {
+  switch (pinnedVersion ?? 'major') {
+  case 'none':
+  case 'major':
+    return `^${version}`
```

Comment:
Since both `'none'` and `undefined` mean `'major'`, I wonder why `'none'` is added.

Reply:
Because currently "none" creates a different specifier for workspace dependencies with `workspace:` and for other regular ranges.